src/ckeditorerror.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * The CKEditor error class.
+ *
+ * All errors will be shortened during the minification process in order to reduce the code size.
+ * Therefore, all error messages should be documented in the same way as those in {@link CKEDITOR.core.log}.
+ *
+ * Read more in the {@link core.log} module.
+ *
+ * @class CKEditorError
+ * @extends Error
+ */
+
+CKEDITOR.define( function() {
+	class CKEditorError extends Error {
```

Comment:
+1 for Fred. Class documentation should describe how to use the class, how it connects with other classes, the idea behind it, etc. Constructor is just like any other method – it accepts stuff and does something. This needs to be documented as well.

Besides, it's really not time  to decide how to document this as we haven't even decided yet with which docs generator we'll go :D. So I'm just following what's already in the code base.